provider/cluster_resource.go

```diff
@@ -427,6 +538,9 @@ func (r *ClusterResource) Read(ctx context.Context, request tfsdk.ReadResourceRe
 		return
 	}
 
+	clusterClient := &ClusterClientImpl{InternalClient: r.collection}
+	clusterUtils := ClusterResourceUtilsImpl{clusterClient: clusterClient}
```

Comment:
It should be created only once when calling to `ClusterResource` constructor and not in each function